tests/python/gaia-ui-tests/gaiatest/visual_components/switch.py

```diff
@@ -0,0 +1,18 @@
+from marionette_driver import Wait
+from gaiatest.visual_components.visual_component import VisualComponent
+
+
+class Switch(VisualComponent):
+    def enable(self):
+        self._toggle_and_verify_state(final_state=True)
+
+    def disable(self):
+        self._toggle_and_verify_state(final_state=False)
+
+    @property
+    def is_checked(self):
+        return self.marionette.execute_script("return arguments[0].wrappedJSObject.checked", [self.root_element])
+
+    def _toggle_and_verify_state(self, final_state):
```

Comment:
I would prefer also a check here that the begin state is not already equal to the final_state, so I guess: Wait(self.marionette).until(lambda m: self.is_checked is not final_state)